packages/xrpl/tools/generateModels.js

```diff
@@ -14,32 +14,26 @@ function readFile(filename) {
 let jsTransactionFile
 
 function processRippledSource(folder) {
-  const sfieldCpp = readFile(
-    path.join(folder, 'src/libxrpl/protocol/SField.cpp'),
+  const sfieldMacroFile = readFile(
+    path.join(folder, '/include/xrpl/protocol/detail/sfields.macro'),
   )
-  const sfieldHits = sfieldCpp.match(
-    /^ *CONSTRUCT_[^\_]+_SFIELD *\( *[^,\n]*,[ \n]*"([^\"\n ]+)"[ \n]*,[ \n]*([^, \n]+)[ \n]*,[ \n]*([0-9]+)(,.*?(notSigning))?/gm,
+  const sfieldHits = sfieldMacroFile.matchAll(
+    /^ *[A-Z]*TYPED_SFIELD *\( *sf([^,\n]*),[ \n]*([^, \n]+)[ \n]*,[ \n]*([0-9]+)(,.*?(notSigning))?/gm,
   )
   const sfields = {}
   for (const hit of sfieldHits) {
-    const matches = hit.match(
-      /^ *CONSTRUCT_[^\_]+_SFIELD *\( *[^,\n]*,[ \n]*"([^\"\n ]+)"[ \n]*,[ \n]*([^, \n]+)[ \n]*,[ \n]*([0-9]+)(,.*?(notSigning))?/,
-    )
-    sfields[matches[1]] = matches.slice(2)
+    sfields[hit[1]] = hit[2]
   }
 
-  const txFormatsCpp = readFile(
-    path.join(folder, 'src/libxrpl/protocol/TxFormats.cpp'),
+  const transactionsMacroFile = readFile(
+    path.join(folder, '/include/xrpl/protocol/detail/transactions.macro'),
   )
-  const txFormatsHits = txFormatsCpp.match(
-    /^ *add\(jss::([^\"\n, ]+),[ \n]*tt[A-Z_]+,[ \n]*{[ \n]*(({sf[A-Za-z0-9]+, soe(OPTIONAL|REQUIRED|DEFAULT)},[ \n]+)*)},[ \n]*[pseudocC]+ommonFields\);/gm,
+  const txFormatsHits = transactionsMacroFile.matchAll(
+    /^ *TRANSACTION\(tt[A-Z_]+ *,* [0-9]+ *, *([A-Za-z]+)[ \n]*,[ \n]*\({[ \n]*(({sf[A-Za-z0-9]+, soe(OPTIONAL|REQUIRED|DEFAULT)(, soeMPT(None|Supported|NotSupported))?},[ \n]+)*)}\)\)$/gm,
   )
   const txFormats = {}
```

Comment:
Question about line 53: Why aren't these three transactions part of the `SubmittableTransaction` type?

`existingLibraryTxs.push('EnableAmendment', 'SetFee', 'UNLModify')`

Reply:
Pseudo-transactions aren't submittable.

https://xrpl.org/docs/references/protocol/transactions/pseudo-transaction-types/pseudo-transaction-types

Reply:
thanks